graph/floyd_warshall.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * @function floydWarshall
+ * @description Compute the shortest path for all pairs of nodes for a graph without negative weight edges. The input graph is a adjacency matrix, where graph[i][j] holds the weight of edges a->b. If the edge does not exist, the value in the matrix is Infinity.
+ * @Complexity_Analysis
+ * Time complexity: O(V^3)
+ * Space Complexity: O(V^2). This space is required to hold the result
+ * @param {number[][]} graph - The graph in adjacency matrix form
+ * @return {number[][]} - A matrix holding the shortest path for each pair of nodes. matrix[i][j] holds the distance of the shortest path (i -> j).
+ * @see https://en.wikipedia.org/wiki/Floyd%E2%80%93Warshall_algorithm
+ */
+export const floydWarshall = (graph: number[][]): number[][] => {
+  let distances = structuredClone(graph);
+  let N = graph.length;
+
+  // For the k'th iteration, we compute all shortest paths with at most k+1 nodes
```

Comment:
This comment seems misleading to me. What Floyd-Warshall does in the `k`-th iteration is try to relax all current shortest paths by going over the `k`-th node. `k` is the index of a node, it's not the max length of the path.

I'm not even sure whether the path lengths are limited by `k` in any way. After all, the algorithm effectively concatenates two paths when it goes over node `k`. This does not guarantee that the resulting path is at most one longer.

TL;DR: The implementation is correct, but the explanation in the comments seems misleading.

Reply:
You're right. What I wrote is incorrect. I rewrote the comments. Hopefully its better.